### ISSUE TITLE
Regression:` bad operand type for unary -:` for `np.sum`

### ISSUE THREAD

**Issue by DerWeh** [PyCQA/pylint#4960]
### Bug description

Pylint doesn't allow `-` infront of `np.sum` anymore.

Minmal example:

```python
# test.py
import numpy as np

result = -np.sum(np.arange(4))
```

With `Pylint` version 3.9.6 everything was fine, after upgrading it is detected as error.

### Command used

```shell
pylint test.py --extension-pkg-whitelist=numpy
```

### Pylint output

```shell
test.py:3:9: E1130: bad operand type for unary -: NoneType (invalid-unary-operand-type)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 2.10.2
astroid 2.7.3
Python 3.7.3 (default, Mar 27 2019, 22:11:17) 
[GCC 7.3.0]
```

### OS / Environment

```shell
Linux version 4.15.0-153-generic (buildd@lgw01-amd64-004) (gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04)) #160-Ubuntu SMP Thu Jul 29 06:54:29 UTC 2021
```

### Additional dependencies

```shell
numpy 1.20.0
```

**Comment by DudeNr33** [comment]
Can you please provide information on the OS and version of ``numpy``? 
I cannot reproduce this message with ``pylint`` 2.10.2 and ``astroid`` 2.7.3.

**Comment by DerWeh** [comment]
@DudeNr33 I added the versions: locally I use Ubuntu, Numpy is version 1.20.0.

The error popped up on Travis, where Pylint suddenly fails in spite of not changing the corresponding files.
On Travis the following OS is used:
Description:	Ubuntu 16.04.6 LTS
Codename:	xenial
with 
```shell
pylint-2.10.2-py3-none-any.whl
numpy-1.21.2-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl
```
the test fails, while with
```shell
pylint-2.9.6-py3-none-any.whl
numpy-1.21.1-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl
```
it still worked.

Probably it is not much help/to tedious, but in case e.g. here you find the travis builds:
https://app.travis-ci.com/github/DerWeh/gftools/jobs/535516900

**Comment by DudeNr33** [comment]
I can now confirm that I can reproduce the regression thanks to the Travis log.
However, I have to clone your repo for it, the given minimal example is not enough to trigger the message. 

The ``invalid-unary-operand-type`` check itself relies entirely on the ``UnaryOp.type_errors()`` method from ``astroid``.
I will look into it in more detail tomorrow.

**Comment by DudeNr33** [comment]
The commit that introduced this is https://github.com/PyCQA/astroid/commit/8049834c518adae06f6848f01bcba33f49e5cdf5.

**Comment by DerWeh** [comment]
Sorry, my bad. I have a configuration file, which I forgot about. The project contains a `.pylintrc`, that is why it is working.

If you use

```shell
pylint test.py --extension-pkg-whitelist=numpy
```

instead, you should be able to reproduce the issue.  I updated the description.

**Comment by DudeNr33** [comment]
Ah, thank you for the clarification! 
That indeed now also triggers this message even for the given minimal example.

@hippo91 it seems like this happens because in ``AstroidBuilder._post_build`` the ``self._manager.visit_transforms(module)`` is no longer called if the package is in the whitelist:

```python
        # Visit the transforms
        if self._apply_transforms:
            if modutils.is_module_name_part_of_extension_package_whitelist(
                module.name, self._manager.extension_package_whitelist
            ):
                return module
            module = self._manager.visit_transforms(module)
        return module
```

**Comment by hippo91** [comment]
@DerWeh sorry for this regression. It seems that commit PyCQA/astroid@8049834 was not so clever...
@DudeNr33 thanks a lot for your investigation.
In fact the option : `--extension-pkg-whitelist=np` doesn't solve the problem. It is identical to not using the option at all, which results in using the astroid's brain in which the `np.sum` function is correctly inferred.
Instead with the `extension-pkg-whitelist=numpy`, the astroid's brain are not used, and astroid load dynamically `numpy`.
I need a bit of time to see how properly allow astroid to load numpy while still using the brains when it needs to.
